app/actions/AuthenticatedActionRefiner.scala

```diff
@@ -46,16 +46,18 @@ trait AuthenticatedActionRefiner { this: ActionRefiner[Request, AuthenticatedReq
     implicit val r: Request[A] = request
 
     authorised(AuthProviders(GovernmentGateway)).retrieve(
-      Retrievals.allEnrolments and Retrievals.credentials
+      Retrievals.allEnrolments and Retrievals.credentials and Retrievals.nino
     ) {
-        case enrolments ~ credentials =>
+        case enrolments ~ credentials ~ nino =>
           credentials match {
             case None =>
               Future.failed(new RuntimeException(s"Could not find credentials"))
 
             case Some(ggCredId) =>
               Future.successful(
-                Right(new AuthenticatedRequest[A](request, enrolments, GGCredId(ggCredId.providerId)))
+                Right(
+                  new AuthenticatedRequest[A](request, enrolments, GGCredId(ggCredId.providerId), Some(Nino(nino.getOrElse(""))))
```

Comment:
it'd make more sense to map into `nino` here so that we get a `None` if `nino` is also `None`